src/main/webapp/resources/js/contexts/AnalysisDetailsContext.js

```diff
@@ -0,0 +1,131 @@
+import React, { useReducer, useContext } from "react";
+
+import {
+  getVariablesForDetails,
+  getAnalysisInputFiles,
+  saveToRelatedSamples,
+  updateAnalysisEmailPipelineResult,
+  updateAnalysis
+} from "../apis/analysis/analysis";
+
+import { showNotification } from "../modules/notifications";
+import { AnalysisContext } from "../contexts/AnalysisContext";
+
+const TYPES = {
+  DETAILS: "ANALYSIS_DETAILS",
+  EMAIL_PIPELINE_RESULT: "UPDATED_EMAIL_PIPELINE_RESULT",
+  PRIORITY: "UPDATED_PRIORITY",
+  SAMPLES: "SAMPLES_DATA",
+  UPDATE_SAMPLES: "UPDATE_SAMPLES"
+};
+
+const reducer = (context, action) => {
+  switch (action.type) {
+    case TYPES.DETAILS:
+      return {
+        ...context,
+        ...action.payload
+      };
+    case TYPES.EMAIL_PIPELINE_RESULT:
+      return { ...context, emailPipelineResult: action.emailPipelineResult };
+    case TYPES.PRIORITY:
+      return { ...context, priority: action.priority };
+    case TYPES.UPDATE_SAMPLES:
+      return { ...context, updateSamples: action.updateSamples };
+    case TYPES.SAMPLES:
+      return {
+        ...context,
+        ...action.payload
+      };
+    default:
+      return;
+  }
+};
+
+const initialContext = {
+  emailPipelineResult: false,
+  workflowName: null,
+  version: null,
+  duration: null,
+  createdDate: null,
+  priority: null,
+  priorities: [],
+  canShareToSamples: false,
+  updatePermission: false,
+  samples: [],
+  sequenceFilePairList: [],
+  sequenceFileSizeList: [],
+  referenceFile: [],
+  updateSamples: false
+};
```

Comment:
Aren't most of these immutable?  Do they need to be in state?

Reply:
The reasoning is that is all passed down as an object from the server. I find this to be a cleaner solution